test/integration/controller/admissionchecks/provisioning/provisioning_test.go

```diff
@@ -726,11 +726,12 @@ var _ = ginkgo.Describe("Provisioning", ginkgo.Ordered, ginkgo.ContinueOnFailure
 			})
 
 			ginkgo.By("Checking if the admission check is still ready and workload is admitted", func() {
-				gomega.Eventually(func(g gomega.Gomega) {
+				gomega.Consistently(func(g gomega.Gomega) {
 					g.Expect(k8sClient.Get(ctx, wlKey, &updatedWl)).To(gomega.Succeed())
 					state := workload.FindAdmissionCheck(updatedWl.Status.AdmissionChecks, ac.Name)
+					g.Expect(state).NotTo(gomega.BeNil())
 					g.Expect(state.State).To(gomega.BeComparableTo(kueue.CheckStateReady))
-					g.Expect(workload.IsAdmitted(&updatedWl)).To(gomega.BeTrue(), "The workload should be admitted")
+					util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, &updatedWl)
```

Comment:
Probably not a good idea to call Eventually inside Consistently, so it might be worth calling `IsStatusConditionTrue` in this case instead.

Reply:
Fixed